### ISSUE TITLE
Attributes created by store_accessor cannot be casted with cast_type

### ISSUE THREAD

**Issue by FX-HAO** [rails/rails#33767]
Let's say we have a model which has a jsonb column named `metadata`.

```ruby
class User < ActiveRecord::Base
  store_accessor :metadata, :foo
  attribute :metadata, ActiveModel::Type::Integer.new
end
```

```ruby
irb> user = User.new(foo: 2.2)
irb> user.foo # should be 2
=> 2.2
```

Attributes created by `store_accessor` should be casted with `cast_type`.
I'll make a PR to fix this.

**Comment by intrip** [comment]
@ghiculescu shall we still keep this open even after https://github.com/rails/rails/pull/33768#issuecomment-805527058 comment ?

**Comment by ghiculescu** [comment]
@rafaelfranca what do you think? I get the impression you'd be fine to see this fixed in Rails (either by integration https://github.com/byroot/activerecord-typedstore or some other fix), but #33768 wasn't a good enough solution. If that's the case I think this should stay open.
